pkg/provisioningv2/rke2/planner/planner.go

```diff
@@ -444,7 +444,7 @@ func (p *Planner) reconcile(controlPlane *rkev1.RKEControlPlane, tokensSecret pl
 					draining = append(draining, entry.Machine.Name)
 				}
 			}
-		} else if !entry.Plan.InSync {
+		} else if !entry.Plan.InSync || entry.Machine.Status.NodeInfo == nil || entry.Machine.Status.NodeInfo.KubeletVersion != controlPlane.Spec.KubernetesVersion {
```

Comment:
```suggestion
		} else if !entry.Plan.InSync || (entry.Machine.Status.NodeInfo != nil && entry.Machine.Status.NodeInfo.KubeletVersion != controlPlane.Spec.KubernetesVersion) {
```

Reply:
I'm thinking this may slow down initial provisioning unnecessarily, but I could be wrong

Reply:
I don't think it will because we won't move forward until the CAPI Machine object reports that the downstream node is ready. That won't happen until this field is populated.

In other words, the machine will be in the `outOfSync` slice instead of the `nonReady` slice in the planner.

Reply:
While that is true for the bootstrap node (because we scrape the joinURL off of the NodeInfo for all except for an etcd-only init node), I'm not sure that is actually the case for the rest of the nodes.

The `nonReady` slice gets rendered and returned into an `errIgnore` rather than an `ErrWaiting`. Looking at the planner, we will move on if there is an `errIgnore`

Reply:
OK. Will update.